src/main/java/javaslang/control/Validation.java

```diff
@@ -0,0 +1,276 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+import javaslang.collection.List;
+
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * An implementation similar to scalaz's <a href="http://eed3si9n.com/learning-scalaz/Validation.html">Validation</a> control.
+ *
+ * <p>
+ * The Validation type is different from a Monad type, it is an applicative
+ * functor. Whereas a Monad will short circuit after the first error, the
+ * applicative functor will continue on, accumulating ALL errors. This is
+ * especially helpful in cases such as validation, where you want to know
+ * all the validation errors that have occurred, not just the first one.
+ * </p>
+ *
+ * <pre>
+ * <code>
+ * Validation.Valid:
+ * Validation&lt;List&lt;String&gt;,Integer&gt; valS = Validation.valid(5);
+ *
+ * Validation.Invalid:
+ * Validation&lt;List&lt;String&gt;,Integer&gt; valE = Validation.invalid(List.of("error1","error2"));
+ * </code>
+ * </pre>
+ *
+ * @param <E> Value type in the case of invalid.
+ * @param <T> Value type in the case of valid.
+ * @author Eric Nelson
+ * @since 2.0.1
+ * @see <a href="http://eed3si9n.com/learning-scalaz/Validation.html">Validation</a>
+ */
+public interface Validation<E,T> extends Kind<Validation<?,?>, E, T>, Applicative<Validation<?,?>, E, T> {
```

Comment:
A general thing: Please let us choose full variable names, e.g. `validation` instead of `v`, same for `v1`, `v2`. I think it increases the readability of our API. It is ok that the lines get slightly longer.